integration-tests-jvm/flink/src/test/java/org/apache/camel/quarkus/component/flink/it/FlinkTest.java

```diff
@@ -16,19 +16,52 @@
  */
 package org.apache.camel.quarkus.component.flink.it;
 
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
 import io.quarkus.test.junit.QuarkusTest;
 import io.restassured.RestAssured;
+import io.restassured.http.ContentType;
 import org.junit.jupiter.api.Test;
 
+import static org.hamcrest.Matchers.greaterThanOrEqualTo;
+
 @QuarkusTest
 class FlinkTest {
 
     @Test
-    public void loadComponentFlink() {
-        /* A simple autogenerated test */
-        RestAssured.get("/flink/load/component/flink")
+    public void dataSetCallback() throws IOException {
+        Path path = Files.createTempFile("fileDataSet", ".txt");
+        String text = "foo\n"
+                + "bar\n"
+                + "baz\n"
+                + "qux\n"
+                + "quux";
+        Files.write(path, text.getBytes(StandardCharsets.UTF_8));
+        RestAssured.given()
+                .contentType(ContentType.TEXT)
+                .post("/flink/dataset/{fileName}", path.toAbsolutePath().toString())
+                .then()
+                .statusCode(200)
+                .and()
+                .body(greaterThanOrEqualTo("5"));
+
+    }
+
+    @Test
+    public void dataStreamCallback() throws IOException {
+        String text = "Hello!!Camel flink!";
+        RestAssured.given()
+                .contentType(ContentType.TEXT)
+                .body(text)
+                .post("/flink/datastream")
                 .then()
-                .statusCode(200);
+                .statusCode(200)
+                .and()
+                .body(greaterThanOrEqualTo("1"));
```

Comment:
I assume we assert `greaterThanOrEqualTo("1")` because Flink is still busy doing work asynchronously? 

Do we know what the length of the file content should be?

If so, it might be better to use awaitility here and just wait until we assert the file size is what we expect.

Reply:
Yes I tried to handle it may be you can looks now!